certifier-osv.md

```diff
@@ -0,0 +1,181 @@
+---
+
+layout: page
+
+permalink: /osv-certifier/
+
+title: "OSV Certifier Documentation"
+
+description: "Guide to using the OSV Certifier in GUAC for vulnerability management"
+
+---
+
+# OSV Certifier Documentation
+
+## Overview
+
+The OSV Certifier component of [GUAC](https://guac.sh) (Graph for Understanding Artifact Composition) integrates with the [OSV (Open Source Vulnerability) database](https://osv.dev) to provide vulnerability insights for open-source dependencies. It enables security risk assessment through vulnerability identification in software dependencies.
+
+## Key Features
+
+-  **Vulnerability Detection**: Scans dependencies using SBOMs and cross-references with OSV's vulnerability catalog
+
+-  **Automated Updates**: Regular synchronization with OSV for current vulnerability data
+
+-  **Comprehensive Reporting**: Structured reports showing vulnerabilities by dependency and severity
+
+
+## Integration Details
+
+### Data Collection Process
+
+1.  **Dependency Matching**:
+
+- Parses SBOM files for package information
+
+- Extracts package names, versions, and ecosystem identifiers
+
+- Validates package metadata format
+
+2.  **API Querying**:
+
+- Queries OSV using standardized package identifiers
+
+- Batch processing for multiple dependencies
+
+- Handles API rate limiting and retries
+
+3.  **Data Correlation**:
+
+- Maps vulnerabilities to GUAC's dependency graph
+
+- Associates vulnerability data with package versions
+
+- Maintains relationship between dependencies
+
+### Covered Ecosystems
+
+The OSV Certifier enables vulnerability detection across several verified package ecosystems, including npm, PyPI, Maven, Go, Cargo, and NuGet. Additionally, it covers a wide range of ecosystems: AlmaLinux, Alpine, Android, Bitnami, crates.io, Curl, Debian GNU/Linux, Git (for C/C++), GitHub Actions, Haskell, Hex, the Linux kernel, OSS-Fuzz, Packagist, Pub, Python (CRAN and Bioconductor), Rocky Linux, RubyGems, SwiftURL, and Ubuntu OS.
+
+### Feature Support
+
+**Supported Features:**
+
+- Public vulnerability data from OSV’s database
+
+- Dependency version mapping against known vulnerabilities
+
+- Analysis of version ranges
+
+- Package identification through [PURL](https://github.com/package-url/purl-spec)
+
+- Severity classification using [CVSS](https://www.first.org/cvss/)
+
+**Unsupported Features:**
+
+- Detection of private vulnerabilities
+
+- Non-OSV-covered ecosystems
+
+- Binary vulnerability scanning
+
+- Custom vulnerability feeds
+
+## Available Options
+
+### Usage
+
+Basic command syntax:
+```bash
+guacone  certifier  osv [options]
+```
+
+
+### Flags
+| Flag | Description | Default |
+|--------------------------------|---------------------------------------------------------------------------------------------|----------------------------------|
```

Comment:
I think you need a blank line between the heading and the start of the table here. It renders fine in the GitHub viewer, but if you look at the CI build, it's not rendered correctly: https://deploy-preview-169--resonant-wisp-1a517a.netlify.app/osv-certifier/

Since today is the 31st, I'll add the hacktoberfest-approved label in case you aren't able to get this finished today.

Reply:
I am doing it just now.

Reply:
```
### Flags

| Flag | Description | Default |
|--------------------------------|---------------------------------------------------------------------------------------------|----------------------------------|
|  `--certifier-batch-size int`  | Sets the batch size for pagination query for the certifier. | 60000 |
|  `--certifier-latency string`  | Sets artificial latency on the certifier (e.g., m, h, s, etc.). | Not enabled (empty) |
|  `-h, --help`  | Help for osv |  |
|  `-l, --last-scan int`  | Hours since the last scan was run; if not set, runs on all packages/sources. | 4 |

```

Maybe now it will work fine

Reply:
**Check this out it's fixed now**
![image](https://github.com/user-attachments/assets/94a87648-d790-4bf2-a9c9-3458fbe87313)
![image](https://github.com/user-attachments/assets/e2745246-816e-40f2-a95a-293ef2fa6ba1)